### PR TITLE
Update to remove old kafka modules

### PR DESCRIPTION
**Pull request by rob3000**
Removing old node modules and use kafkaJS as primary source and removed some old config. 

Suggestions/comments welcome 😄 

Next i'd like to update some of the documentation thoughts on using something like: https://docusaurus.io/

### CODE REVIEW

**Comment by krystianity** [comment]
Looking good @rob3000 I am on the road currently, going to have a look at this on the weekend.

File: lib/kafkajs/JSConsumer.js
Comment (krystianity) [inline lib/kafkajs/JSConsumer.js]: the whole noptions block was taken from the native kafka client, which they abbreviated from the native xml kafka options. I would suggest sticking to the same config names (meaning keeping the metadata. prefix) to prevent frustration from people updating to the latest version of our lib

File: yarn.lock
Comment (krystianity) [inline yarn.lock:123]: i am not 100% sure if yarn is they way to go anymore, npm has done some improvements in the past. What do you think? Should we switch from yarn to npm lock files?

File: lib/Sinek.js
Comment (krystianity) [inline lib/Sinek.js:23]: we should definitely mention this clearly in the change log, we removed the depracted client versions and explicity removed the native client from the whole lib

**Review by krystianity — CHANGES_REQUESTED** [review]
Not really a change request, more like suggestions and opinions ;)

File: yarn.lock
Comment (rob3000) [reply yarn.lock:123]: Just opened an issue: https://github.com/nodefluent/node-sinek/issues/155 to discuss this further

Comment (krystianity) [reply yarn.lock:123]: Roger

**Comment by rob3000** [comment]
@krystianity updated based on the comments you made. I've also converted to typescript. Currently admin client is failing but feel free to review :)

**Comment by rob3000** [comment]
@krystianity This is now ready for review! 🥳

**Comment by krystianity** [comment]
Awesome, will review tonight - you are really going full speed here 👍 ❤️

File: index.d.ts
Comment (krystianity) [inline index.d.ts]: this is still empty, however since the code base is now transpiled anyway we should not need the definition file anymore right?

File: src/lib/kafkajs/index.ts
Comment (krystianity) [inline src/lib/kafkajs/index.ts]: minimal stuff, lots of files missing ending new line, you should probably change this in your editors settings :D

**Comment by krystianity** [comment]
looking good so far, however the amount of changes is quite large so its quite difficult to approve this with 100% confidence, given the major changes we are doing anyway, we will probably have to run some in the fields tests for lag status (health and analytics) - testing this with kafka-streams will probably also be a good validation of the changes.

File: index.d.ts
Comment (rob3000) [reply index.d.ts]: yeah correct. Thought i had removed it. Will remove now 😄

**Comment by rob3000** [comment]
agreed, there are lots of changes. i guess if we merge to master we can test kafka streams from the sinek's master branch which would give us better confidence?

File: src/lib/kafkajs/index.ts
Comment (rob3000) [reply src/lib/kafkajs/index.ts]: 🤔 i have a feeling that eslint --auto-fix removed them as i normally leave a new line. Will look into and fix up.

Comment (rob3000) [reply src/lib/kafkajs/index.ts]: Nope, it was actually my editor 🤭

**Comment by krystianity** [comment]
@rob3000 didnt release you were waiting for my approval to merge this, lgtm ;)

**Pull request merged**
